core/src/com/unciv/ui/screens/worldscreen/unit/actions/UnitActions.kt

```diff
@@ -158,11 +158,41 @@ object UnitActions {
                 GUI.getMap().setCenterPosition(unit.getMovementDestination().position, true)
             })
         }
-
+        addEscortAction(unit)
         addSwapAction(unit)
         addDisbandAction(unit)
     }
 
+    private suspend fun SequenceScope<UnitAction>.addEscortAction(unit: MapUnit) {
+        // Air units cannot escort
+        if (unit.baseUnit.movesLikeAirUnits()) return
+
+        val worldScreen = GUI.getWorldScreen()
+        val selectedUnits = worldScreen.bottomUnitTable.selectedUnits
+        if (selectedUnits.size == 2) {
+            val tile = selectedUnits.first().getTile()
+            if (selectedUnits.last().getTile() != tile) return
+            if (selectedUnits.any { it.baseUnit.movesLikeAirUnits() }) return
+        } else if (selectedUnits.size == 1) {
+        } else {
+            return
+        }
```

Comment:
```suggestion
        } else if (selectedUnits.size != 1) return
```

Reply:
Although I'm not sure what exactly you're getting at here, what is escort with 1 unit?

Reply:
We usually only have one unit selected. We need to get the other unit on the tile to get the two units. If we select two units with multi-select, this code allows us to still put them into a formation.